### PR TITLE
274 messaging error bug

### PR DESCRIPTION
**Pull request by Sharjanan**
-Fixed chat list
-User can chat only with connected users

### CODE REVIEW

File: frontend/src/components/DirectMessage/Messaging.js [hunk 1]
```diff
@@ -112,6 +127,47 @@ const Message = () => {
...
+      console.log(ids);
+    });
+  }, []);
+
+/** 
+   * Accepts a function that contains imperative, possibly effectful code.
+   * @param effect — Imperative function that can return a cleanup function
+   * @param deps — If present, effect will only activate if the values in the list change.
+  */
+useEffect(() => {
```
Comment (Chovv) [inline frontend/src/components/DirectMessage/Messaging.js:148]: I see that filtered the users based on the existing connections in the DB.

**Review by fatema-a — APPROVED** [review]
great additions now users only see chats with people they are connected too instead of list with whole user list, make functionality better

File: frontend/src/components/DirectMessage/Messaging.js [hunk 2]
```diff
@@ -281,7 +337,7 @@ const Message = () => {
             <div style={{ overflowY: "scroll", height: "500px" }}>
               {/* Maps through all the users in the 'users_information' array, 
               except the current user, and renders a 'user-tab' div for each */}
-              {users_information.map(
+              {connections.map(
```
Comment (Chovv) [inline frontend/src/components/DirectMessage/Messaging.js:340]: The feature works, and I can only see the DMs with people I'm connected with. Nice work!

**Review by Chovv — APPROVED** [review]
The DM filtering feature works as intended, and effectively closes the DM bug we had previously.

**Pull request merged**